pulser-core/pulser/sequence/_schedule.py

```diff
@@ -102,25 +147,22 @@ def get_samples(self, modulated: bool = False) -> ChannelSamples:
             )
             phase[t_start:t_end] += pulse.phase
             tf = s.tf
-            if modulated:
```

Comment:
I got rid of the option of getting the channel samples modulated directly from `_ChannelSchedule.get_samples()` because it was giving inconsistent results. 
The main issue was that modulation followed by extension (ie increasing the length of the samples) is different then extension follow by modulation; I made it so modulation allows followed extension when creating the `SequenceSamples`.
The changes are isolated in ea3b3a73464879dac299d4974586fec085672b75.

As for this case in particular, the `Pulse.fall_time()` is zero when there is no modulation bandwidth specified, so there's no need for the `if` statement